Content.Server/Clothing/Systems/SkatesSystem.cs

```diff
@@ -0,0 +1,57 @@
+using Content.Shared.Clothing;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Movement.Components;
+using Content.Shared.Movement.Systems;
+using Content.Server.Damage.Components;
+
+namespace Content.Server.Clothing;
+
+public sealed class SkatesSystem : EntitySystem
+{
+    [Dependency] private readonly MovementSpeedModifierSystem _move = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SkatesComponent, GotEquippedEvent>(OnGotEquipped);
+        SubscribeLocalEvent<SkatesComponent, GotUnequippedEvent>(OnGotUnequipped);
+    }
+
+    public void OnGotUnequipped(EntityUid uid, SkatesComponent component, GotUnequippedEvent args)
+    {
+        if (args.Slot == "shoes")
+        {
+            if (TryComp<MovementSpeedModifierComponent>(args.Equipee, out var mover))
+            {
+                mover.Friction = 20f;
+                mover.FrictionNoInput = null;
+                mover.MinimumFrictionSpeed = 0.005f;
+                _move.RefreshMovementSpeedModifiers(args.Equipee);
+            }
+
+            if (TryComp<DamageOnHighSpeedImpactComponent>(args.Equipee, out var impact))
+            {
+                impact.MinimumSpeed = 20f;
+            }
+        }
```

Comment:
Why are you setting the fields and also refreshing movespeed modifiers? None of the fields are networked either and physics needs to be 100% predicted.

Reply:
sorted